src/AppLegacy.php

```diff
@@ -0,0 +1,399 @@
+<?php
+
+// Copyright (C) 2010-2024, the Friendica project
+// SPDX-FileCopyrightText: 2010-2024 the Friendica project
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+namespace Friendica;
+
+use DateTimeZone;
+use Exception;
+use Friendica\App\BaseURL;
+use Friendica\App\Mode;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\Config\ValueObject\Cache;
+use Friendica\Core\L10n;
+use Friendica\Core\PConfig\Capability\IManagePersonalConfigValues;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Core\System;
+use Friendica\Core\Theme;
+use Friendica\Database\Database;
+use Friendica\Network\HTTPException\InternalServerErrorException;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Strings;
+
+/**
+ * Helper for our main application structure for the life of this page.
+ *
+ * Primarily deals with the URL that got us here
+ * and tries to make some sense of it, and
+ * stores our page contents and config storage
+ * and anything else that might need to be passed around
+ * before we spit the page out.
+ *
+ */
+final class AppLegacy implements AppHelper
```

Comment:
Wouldn't here be a paradise for new unit-tests? :)
You could define every kind of expected content here. In particular on the `interface` level, making sure that other kind of implementations will follow the given expectations

Reply:
I would not create unit tests for `AppLegacy` because I extracted mostly getter and setter from `App` that has to be moved again to other places. I don't think it is worth it.

But the usage of `AppHelper` in other places could later be used to mock the `App` class.